src/presence_analyzer/static/js/mean_time.js

```diff
@@ -6,38 +6,63 @@ function parseInterval(value) {
 
 (function($) {
     $(document).ready(function(){
-        var $loading = $('#loading');
+        var $loading = $("#loading");
 
-        $.getJSON("/api/v1/users", function(result) {
+        $("#user_id").change(function(){
+            var $selectedUser = $("#user_id").val();
+
+            $.getJSON("api/v1/get_avatar/" + $selectedUser, function(result){
+                var $_img = document.getElementById("avatar");
+                $newImg = new Image;
+                $newImg.src = result["avatar"];
+
+                $newImg.onload = function() {
+                    $_img.src = this.src;
+                }
+            });
+        });
+
+        $.getJSON("/api/v1/users_xml", function(result) {
             var $dropdown = $("#user_id");
             $.each(result, function(item) {
                 $dropdown.append($("<option />").val(this.user_id).text(this.name));
             });
             $dropdown.show();
             $loading.hide();
         });
-        $('#user_id').change(function(){
+        $("#user_id").change(function(){
             var $selectedUser = $("#user_id").val(),
-                $chartDiv = $('#chart_div');
+                $chartDiv = $("#chart_div");
 
             if($selectedUser) {
                 $loading.show();
                 $chartDiv.hide();
 
+                fetch("/api/v1/mean_time_weekday/" + $selectedUser)
+                    .then(function(response) {
+                        if (!response.ok) {
+                            throw Error(response.statusText);
+                        }
+                        return response;
+                    }).catch(function(error) {
+                        alert('User data not found!');
+                    });
```

Comment:
Why? I don't understand of sending call to the same api twice (41, 51). $.getJSON returns a promise on which u can catch fail. Check jQuery docs